components/lib/publicsuffixlist/src/test/java/mozilla/components/lib/publicsuffixlist/PublicSuffixListTest.kt

```diff
@@ -402,4 +402,24 @@ class PublicSuffixListTest {
         // https://github.com/mozilla-mobile/android-components/issues/3540
         assertEquals("1", publicSuffixList.getPublicSuffix("127.0.0.1").await())
     }
+
+    @Test
+    fun `Verify getPublicSuffix return value is unaffected by trailing slash`() {
+        fun verifySamePublicSuffix(domain: String) {
+            runBlocking {
+                val withSlash = publicSuffixList.getPublicSuffix("$domain/").await()!!
+                val withoutSlash = publicSuffixList.getPublicSuffix(domain).await()!!
+
+                assertEquals(withSlash, "$withoutSlash/")
+            }
+        }
+
+        listOf(
+            "https://www.mozilla.org", // passes
+            "http://web.security.plumbing", // passes
+            "https://mozfreddyb.github.io" // fails
+                // https://mozfreddyb.github.io -> “io”
+                // https://mozfreddyb.github.io/ -> “github.io/”
+        ).forEach(::verifySamePublicSuffix)
```

Comment:
Please note that the expected input to `PublicSuffixList" is a domain/host and not a URL and as the kdoc mentions it only applies the rules of the public suffix list and does not do any domain validations.

However of course we could consider adding an extension method that takes a URL and does the splitting etc. for you.

Reply:
I think we can probably close this issue then.  Would you mind if I submitted a PR adding that note to the method kdocs?  That's where I would typically look for information about expected inputs.

Reply:
To be clear, I do recognize that I _should_ have gotten that from the `domain` param name and the example input -> output pairs.  That's on me.  But I think it might help to make it more explicit on method docs.

Reply:
Yeah, I agree. I think it would be good to make this more explicit!